protodroid/src/main/java/id/lukasdylan/grpc/protodroid/internal/viewmodel/MainViewModel.kt

```diff
@@ -28,15 +42,55 @@ internal class MainViewModel(private val repository: InternalProtodroidRepositor
             repository.deleteAllData()
         }
     }
+
+    /**
+     * Filter data with no duplicates or successful responses by default
+     */
+    private suspend fun getFilteredData(): Flow<List<ProtodroidDataEntity>> {
```

Comment:
How about we using combine from flow, so no need to manually call the repository.
https://betterprogramming.pub/learn-how-to-combine-kotlin-flows-317849a71d3e

Reply:
So the filterList will be wrapped as Flow, and then used emit regarding changes of the filterList

Reply:
I like the idea, but do you have a suggestion on how to do this, we'll have to call `repository.fetchAllData()` at least once right?

Reply:
I think we can add on this block 
`init {
        viewModelScope.launch {
            getFilteredData().collect {
                _dataResponse.emit(it)
            }
        }
    }`
    
   become
   `init {
        viewModelScope.launch {
            repository.fetchAllData().combine(filterListFlow) { dataLogList, filterList ->
                // logic to filter and distinct 
            }.collect { finalList ->
                _dataResponse.emit(finalList)
            }
        }
    }`

Reply:
Let me know if this approach didn't work

Reply:
Nice, I like it. Works well too